File: Chapter 2/MultiArmedBandit.m
classdef MultiArmedBandit
    properties
        n_arms
        arm_values
    end
    
    methods
        function obj = MultiArmedBandit(n_arms)
            obj.n_arms = n_arms;
            obj.arm_values = normrnd(0, 1, [1, n_arms]);
        end
        
        function reward = pull(obj, arm)
            reward = normrnd(obj.arm_values(arm), 1);
        end
    end
end

% Example usage
bandit = MultiArmedBandit(10);
reward = bandit.pull(3);
disp(['Reward received: ', num2str(reward)]);

File: Chapter 2/MultiArmedBandit.py
import numpy as np

class MultiArmedBandit:
    def __init__(self, n_arms):
        self.n_arms = n_arms
        self.arm_values = np.random.normal(0, 1, self.n_arms)

    def pull(self, arm):
        reward = np.random.normal(self.arm_values[arm], 1)
        return reward

# Example usage
bandit = MultiArmedBandit(10)
reward = bandit.pull(3)
print(f"Reward received: {reward}")